lighthouse-core/runner.js

```diff
@@ -59,27 +46,49 @@ class Runner {
         data: sentryContext && sentryContext.extra,
       });
 
-      // If the URL isn't https and is also not localhost complain to the user.
-      if (parsedURL.protocol !== 'https:' && parsedURL.hostname !== 'localhost') {
-        log.warn('Lighthouse', 'The URL provided should be on HTTPS');
-        log.warn('Lighthouse', 'Performance stats will be skewed redirecting from HTTP to HTTPS.');
-      }
-
-      // canonicalize URL with any trailing slashes neccessary
-      const requestedUrl = parsedURL.href;
-
       // User can run -G solo, -A solo, or -GA together
       // -G and -A will run partial lighthouse pipelines,
       // and -GA will run everything plus save artifacts to disk
 
       // Gather phase
       // Either load saved artifacts off disk or from the browser
       let artifacts;
+      let requestedUrl;
       if (settings.auditMode && !settings.gatherMode) {
         // No browser required, just load the artifacts from disk.
         const path = Runner._getArtifactsPath(settings);
         artifacts = await assetSaver.loadArtifacts(path);
+        requestedUrl = artifacts.URL.requestedUrl;
+
+        if (!requestedUrl) {
+          throw new Error('Cannot run audit mode on empty URL');
+        }
+        if (opts.url && opts.url !== requestedUrl) {
+          throw new Error('Cannot run audit mode on different URL');
+        }
       } else {
+        // save the requestedUrl provided by the user
+        const rawRequestedUrl = opts.url;
+        if (typeof rawRequestedUrl !== 'string' || rawRequestedUrl.length === 0) {
+          throw new Error(`You must provide a url to the runner. '${rawRequestedUrl}' provided.`);
+        }
+
+        let parsedURL;
+        try {
+          parsedURL = new URL(opts.url);
+        } catch (e) {
+          throw new Error('The url provided should have a proper protocol and hostname.');
+        }
+
+        // If the URL isn't https and is also not localhost complain to the user.
+        if (parsedURL.protocol !== 'https:' && parsedURL.hostname !== 'localhost') {
+          log.warn('Lighthouse', 'The URL provided should be on HTTPS');
+          log.warn('Lighthouse', 'Performance stats might be skewed redirecting to HTTPS.');
+        }
+
+        // canonicalize URL with any trailing slashes neccessary
```

Comment:
this comment is just weird now. Add an `if`? Drop the `necessary`? Something else?

Reply:
how about .. move this comment to  L77 and drop the use of `parsedURL` and s/parsedURL/requestedUrl/ ?

Reply:
done